### DISCUSSION TITLE
Get radio input value as number or boolean in uncontrolled component

### DISCUSSION THREAD

**Discussion by ImAbdullahJan** [react-hook-form/react-hook-form#4026] [Q&A] [answered]
I am currently using uncontrolled components throughout my app and use react-hook-form to manage form state. 

I have 2 scenarios related to input [type=radio]. 

In 1 scenario, I need a response as the number and in another, I need a response as a boolean. 

I see solutions with the Controller component and controlled input. but my components are uncontrolled. 

Please suggest any solution. 

CSB link: [https://codesandbox.io/s/react-hook-form-validation-valueasnumber-qzzx5](https://codesandbox.io/s/react-hook-form-validation-valueasnumber-qzzx5)

**Comment by bluebill1049** [comment, marked as answer]
`valueAsNumber` is only applicable for text input.

**Reply by ImAbdullahJan** [reply]
Any solution for uncontrolled components?

**Reply by bluebill1049** [reply]
I think you will have to build a controlled group component, we are following the HTML standard, `valueAsNumber` doesn't support radio input.

**Reply by ImAbdullahJan** [reply]
Hi @bluebill1049, 

Any example in docs related to the controlled group component?

**Reply by bluebill1049** [reply]
we don't, but Controller is what you need to get it working: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r
